packages/medusa/src/services/product-variant.ts

```diff
@@ -362,7 +362,6 @@ class ProductVariantService extends TransactionBaseService {
           await this.setCurrencyPrice(variantId, price)
         }
       }
-
       await moneyAmountRepo.remove(obsoletePrices)
```

Comment:
question: if performance is a concern, why don't we delete obsolete prices directly with a query instead of fetching all the records just to pass them to `moneyAmountRepo.remove`

Reply:
I agree that we could have a method on the moneyAmountRepo which would be `deleteVariantPricesNotIn` which would create a query with the ORM to remove them directly instead of retrieving and then delete

Reply:
I can't find any other use of `findVariantPricesNotIn`, so I believe we should be able to safely remove directly as part of the query @kasperkristensen